services/frontend/src/i18n/tools/validationHelperFunctions.js

```diff
@@ -148,84 +148,120 @@ const getFilesInDirectory = async (dir, ext) => {
-  ignoreKeys
+  blacklistedKeys
 ) => {
   const unusedKeys = [];
 
   // get all files in directory and flatten into one variable
   let files = await Promise.all(
     ext.map((extension) => getFilesInDirectory(dir, extension))
   );
+
   files = _.flatten(files);
   const filesContent = await Promise.all(
     files.map(async (i) => fs.readFile(i).then((buffer) => buffer.toString()))
   );
 
-  // function to search for key in file content
-  const searchContentForKey = async (key, contentToSearch) =>
-    contentToSearch.some(
+  searchableKeys.map(async (key) => {
+    const result = filesContent.some(
       (content) =>
         _.includes(content, `"${key}"`) ||
         _.includes(content, `'${key}'`) ||
         _.includes(content, `\`${key}\``) ||
-        _.includes(ignoreKeys, key)
+        _.includes(blacklistedKeys, key)
     );
+    if (!result) unusedKeys.push(key);
+  });
+
+  if (unusedKeys.length) {
+    console.error(
+      `${unusedKeys.length} keys that are unused in the project found`,
+      unusedKeys
+    );
+    console.error("findUnusedTranslations check: FAIL\n");
+  } else {
+    console.error("All keys in en_CA and fr_CA files are used in the project!");
+    console.error("findUnusedTranslations check: SUCCESS\n");
+  }
 
-  await Promise.all(
-    searchableKeys.map(async (key) => {
-      const found = await searchContentForKey(key, filesContent);
-      if (!found) {
-        unusedKeys.push(key);
-      }
-    })
-  );
   return unusedKeys;
 };
 
-/**
- * A test to fined unused translations in the entire project
- *
- * @param {object} enTranslations List of english translations
- * @param {object} frTranslations List of english translations
- * @param {string[]} blacklistedKeys List of keys to be ignored
- */
-const findUnusedTranslations = async (
-  enTranslations,
-  frTranslations,
+const findMissingTranslations = async (
+  dir,
+  ext,
+  searchableKeys,
   blacklistedKeys
 ) => {
-  const enKeys = Object.keys(enTranslations);
-  const frKeys = Object.keys(frTranslations);
-  let unusedKeys = {};
+  // get all files in directory and flatten into one variable
+  let files = await Promise.all(
+    ext.map((extension) => getFilesInDirectory(dir, extension))
+  );
 
-  const allKeys = _([...enKeys, ...frKeys])
-    .uniq()
-    .sort()
-    .value();
+  files = _.flatten(files);
+  const filesContent = await Promise.all(
+    files.map(async (i) => fs.readFile(i).then((buffer) => buffer.toString()))
+  );
+  const valuesMissing = [];
+  const patternIntl =
+    /intl\.formatMessage(?:.|\n)*?\((?:.|\n)*?({.*?id.*?})(?:.|\n)*?\)/gs;
+  const patternDiv = /<FormattedMessage id=['|"|`](?:.|\n)*?['|"|`]/gs;
+
+  const getValue = (data) =>
+    _.compact(
+      data.map((i) => {
+        const value = i.replace(/,|\s/g, "");
+        if (value.match(/["|`|'].*['|"|`]/)) {
+          const cleanedValue = value
+            .match(/["|`|'].*['|"|`]/)[0]
+            .replace(/['|"|`]+/g, "");
+
+          if (
```

Comment:
maybe we should move this into a separate function. let me know what you think :)